include/seqan3/core/metafunction/range.hpp

```diff
@@ -71,7 +71,7 @@ namespace seqan3
  * \tparam t The type to operate on.
  */
 template <std::ranges::Range rng_t>
-using iterator_t = ranges::iterator_t<rng_t>;
+using iterator_t = decltype(ranges::begin(std::declval<rng_t &>()));
```

Comment:
Does this have to be `std::ranges::begin` or can it just be `ranges::begin`?

Reply:
After your PR is merged it should be `std::ranges::begin`, but right now that wouldn't work.